spec/v1.0-draft/resource_descriptor.md

```diff
@@ -0,0 +1,139 @@
+# ResourceDescriptor field type specification
+
+Version v1.0-draft
+
+A size-efficient description of any mutable or immutable software artifact
+or resource.
+
+## Schema
+
+```json
+{
+  "name": "<NAME>",
+  "uri": "<RESOURCE URI>",
+  "digest": { "<ALGORITHM>": "<HEX VALUE>", ... },
+  "representation": { /* object */ },
+  "downloadLocation": "<RESOURCE URI>"
+  "mediaType": "<MIME TYPE>",
+  "annotations": { /* object */ }
+}
+```
+
+## Fields
+
+Though all fields are optional, a ResourceDescriptor MUST specify at least
+one of `uri` or `digest` at a minimum.
+
+`name` _string, optional_
+
+> Human-readable identifier to distinguish the resource or artifact locally.
+> The semantics are up to the producer and consumer.
+
+`uri` _[ResourceURI], optional_
+
+> A URI used to identify the resource or artifact. When a `digest` cannot be
+> computed for the resource or artifact, the producer SHOULD set this field.
+
+`digest` _[DigestSet], optional_
+
+> A set of crypographic digests of the contents of the resource or artifact.
+> When known, the producer SHOULD set this field to denote an immutable
+> artifact or resource. The producer and consumer must agree on acceptable
+> algorithms.
+
+`representation` _object, optional_
+
+> A representation of the contents of the resource or artifact.
+> The producer MAY use this field in scenarios where including the contents
+> of the resource/artifact directly in the attestation is deemed more
+> efficient for consumers than providing a reference to another location.
+>
+> The producer and consumer must agree on the semantics and acceptable
+> formats for the `representation` object.
```

Comment:
I think this would be more clear as `value` with type `bytes` (encoded as base64). Otherwise you cannot digest it. (Same comment as https://github.com/slsa-framework/slsa/issues/649#issuecomment-1454078960.)

Would also be good to RECOMMEND that the size be kept small.

Reply:
Yes, makes perfect sense. I felt the name `value` would be too general, which is why I had initially gone with the slightly more specific (IMO) `representation` (akin to the `repr()` function in python). But, I definitely agree that base64 is the better choice, so the name `representation` also doesn't really make sense. Could we just go with `content` since that's really what the field is for?

Reply:
`content` sounds good to me! That's much better than `value`.

Regarding "small", I had the same thought. I think it's a judgement call but haven't been able to think of more specific guidelines.

Reply:
Suggestion for 'small', "SHOULD be less than 1KB".